builders/interact/ethereum-api/libraries/web3py.md

```diff
@@ -0,0 +1,264 @@
+
+```bash
+touch increment.py reset.py
+```
+
+Open the `increment.py` file and take the following steps to create the script:
+
+1. Add imports, including Web3.py and the ABI of the `Incrementer.sol` contract
+2. [Set up the Web3 provider](#setting-up-the-web3py-provider)
+3. Define the `account_from`, including the `private_key`, the `contract_address` of the deployed contract, and the `value` to increment by. The private key is required to sign the transaction. **Note: This is for example purposes only. Never store your private keys in a Python file**
+4. Create a contract instance using the `web3.eth.contract` function and passing in the ABI and address of the deployed contract
+5. Build the increment transaction using the contract instance and passing in the value to increment by. You'll then use the `build_transaction` function to pass in the transaction information including the `from` address and the `nonce` for the sender. To get the `nonce` you can use the `web3.eth.get_transaction_count` function
+6. Sign the transaction using the `web3.eth.account.sign_transaction` function and pass in the increment transaction and the `private_key` of the sender
+7. Using the signed transaction, you can then send it using the `web3.eth.send_raw_transaction` function and wait for the transaction receipt by using the `web3.eth.wait_for_transaction_receipt` function
+
+```python
+--8<-- 'code/ethereum-api/web3py/increment.py'
+```
+
+To run the script, you can enter the following command in your terminal:
+
+```bash
+python3 increment.py
+```
+
+If successful, the transaction hash will be displayed in the terminal. You can use the `get.py` script alongside the `increment.py` script to make sure that value is changing as expected:
+
+![Increment Contract Web3py](/images/builders/interact/ethereum-api/web3py/web3py-5.png)
+
+Next you can open the `reset.py` file and take the following steps to create the script:
+
+1. Add imports, including Web3.py and the ABI of the `Incrementer.sol` contract
+2. [Set up the Web3 provider](#setting-up-the-web3py-provider)
+3. Define the `account_from`, including the `private_key`, and the `contract_address` of the deployed contract. The private key is required to sign the transaction. **Note: This is for example purposes only. Never store your private keys in a Python file**
+4. Create a contract instance using the `web3.eth.contract` function and passing in the ABI and address of the deployed contract
+5. Build the reset transaction using the contract instance. You'll then use the `build_transaction` function to pass in the transaction information including the `from` address and the `nonce` for the sender. To get the `nonce` you can use the `web3.eth.get_transaction_count` function
```

Comment:
Same problem as the previous script (at least on my computer). Solved it changing the code to:

checksum_address = Web3.to_checksum_address(account_from['address'])
reset_tx = Incrementer.functions.reset().build_transaction(
    {
        'from': checksum_address,
        'nonce': web3.eth.get_transaction_count(checksum_address),
    }
)